### PR TITLE
Fix duplicate rev namespace in injector list

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
The webhook can contains several webhooks that is matching the namespace, and currently the namespace may be duplicated: 
```
istioctl x injector list                    
...

NAMESPACES          INJECTOR-HOOK              ISTIO-REVISION SIDECAR-IMAGE
...
hello               istio-revision-tag-default default        hanxiaop/proxyv2:1.21.0-1226
...
hello               istio-revision-tag-default default        hanxiaop/proxyv2:1.21.0-1226
```

this is to remove the duplications per revisioned webhook.

### CODE REVIEW

File: istioctl/pkg/injector/injector-list.go [hunk 1]
```diff
@@ -229,19 +229,23 @@ func getInjectedRevision(namespace *corev1.Namespace, hooks []admitv1.MutatingWe
...
 		}
 
 		for _, namespace := range namespaces {
 			if nsSelector.Matches(api_pkg_labels.Set(namespace.Labels)) {
-				retval = append(retval, namespace)
+				matched[namespace.Name] = namespace
 			}
 		}
 	}
+	retval := make([]corev1.Namespace, 0)
```
Comment (wulianglongrd) [inline istioctl/pkg/injector/injector-list.go:245]: nit: `retval := make([]corev1.Namespace, 0, len(matched))`.

File: istioctl/pkg/injector/injector-list.go [hunk 2]
```diff
@@ -229,19 +229,23 @@ func getInjectedRevision(namespace *corev1.Namespace, hooks []admitv1.MutatingWe
...
 		nsSelector, err := metav1.LabelSelectorAsSelector(webhook.NamespaceSelector)
 		if err != nil {
-			return retval
+			return nil
 		}
 
 		for _, namespace := range namespaces {
 			if nsSelector.Matches(api_pkg_labels.Set(namespace.Labels)) {
-				retval = append(retval, namespace)
+				matched[namespace.Name] = namespace
```
Comment (wulianglongrd) [inline istioctl/pkg/injector/injector-list.go:241]: I think it might be better to use `set` to prevent duplication.

Comment (hanxiaop) [reply istioctl/pkg/injector/injector-list.go]: Sounds good, fixed

File: istioctl/pkg/injector/injector-list.go [hunk 3]
```diff
@@ -253,12 +256,7 @@ func getPods(ctx context.Context, client kube.CLIClient) (map[resource.Namespace
 		return retval, err
 	}
 	for _, pod := range pods.Items {
-		podList, ok := retval[resource.Namespace(pod.GetNamespace())]
-		if !ok {
-			retval[resource.Namespace(pod.GetNamespace())] = []corev1.Pod{pod}
-		} else {
-			retval[resource.Namespace(pod.GetNamespace())] = append(podList, pod)
-		}
+		retval[resource.Namespace(pod.GetNamespace())] = append(retval[resource.Namespace(pod.GetNamespace())], pod)
```
Comment (ymesika) [inline istioctl/pkg/injector/injector-list.go:259]: Nice simplification

**Review by ymesika — APPROVED** [review]
LGTM

**Review by GregHanson — APPROVED** [review]
lgtm

**Pull request merged**